### PR TITLE
[FIX] fix float_round rounding off error

### PR DESCRIPTION
**Pull request by milwell**
Description of the issue/feature this PR addresses:
float_round rounds off with error on cents. This is specifically a problem when doing Payslip rules.
Example: we have 3 rules: Net Pay, Adjustment, Final Pay
Net Pay  = 1906.68
Adjustment = -1140
Final Pay  = Net Pay + Adjustment  = 766.67 , (should be 766.68  since 1906.68 - 1140 = 766.68)

upon checking float_round seems to be having errors using epsilon value. 

Current behavior before PR:

float_round(1906.6749999999997,precision_digits=2)
result is 1906.68
float_round(766.6749999999997,precision_digits=2)
result is 766.6700000000001

Desired behavior after PR is merged:

float_round(1906.6749999999997,precision_digits=2)
result is 1906.67
float_round(766.6749999999997,precision_digits=2)
result is 766.67

Although decimal part is not 68 it will be ok since  .6749999 is .67 when rounded off

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mart-e** [comment]
Hello,

Thanks for the PR but this most likely incorrect.
Check the build, this PR breaks a lot of rounding tests https://runbot.odoo.com/runbot/build/47871687

If you are going to modify float round, be sure to add some new tests for the case you are fixing.
Your issue is probably a missing `float_repr` not an issue in `float_round`.

TL;DR: use `float_repr` when you want to display a value, use `float_compare` to compare two floats. Doing something like
```
>>> x = float_round(y, ...)
>>> x
1.000000000000042 # some very long float that looks badly rounded
```
is **not** a correct way to check the rounded value

**Comment by milwell** [comment]
This error is happening in the Payroll module, module hr_payroll , file hr_payslip.py method: _get_payslip_lines.
The method returns an array of value dicts and that contains the long numbers such as 1906.6749999999997 which is
saved to payslip rule computation as monetary with 2 decimal places, I'm assuming the monetary field uses float_round.

I don't have control over these modules.  Or should I just file a support ticket?

**Comment by milwell** [comment]
I'm closing this since I have found a workaround on the Salary Rules, I used float_round inside the Salary Rules.

**Comment by mart-e** [comment]
> I don't have control over these modules. Or should I just file a support ticket?

If you can reproduce this in the standard payroll modules, yes you can open a support ticket. But if you have a workaround, it's even faster :)
